test/resource_test.rb

```diff
@@ -142,6 +142,26 @@ def test_acquire_with_fork
     end
   end
 
+  def test_quota_grace_period
```

Comment:
Nice test. Should we also test something where they sleep for longer than the padding time?